src/Compilers/Core/Portable/MetadataReader/MetadataReaderExtensions.cs

```diff
@@ -196,5 +198,25 @@ internal static bool IsPublicNonInterfaceType(this MetadataReader reader, TypeDe
                 reader.StringComparer.Equals(typeDef.Name, typeName) &&
                 reader.StringComparer.Equals(typeDef.Namespace, namespaceName);
         }
+
+        internal static IEnumerable<string> DumpAssemblyReferences(this MetadataReader reader)
+        {
+            return reader.AssemblyReferences.Select(r => reader.GetAssemblyReference(r))
+                .Select(row => $"{reader.GetString(row.Name)} {row.Version.Major}.{row.Version.Minor}");
+        }
+
+        internal static IEnumerable<string> DumpTypeReferences(this MetadataReader reader)
+        {
+            return reader.TypeReferences
+                .Select(t => reader.GetTypeReference(t))
+                .Select(t =>
+                {
+                    var assemblyName = t.ResolutionScope.Kind == HandleKind.AssemblyReference
+                        ? reader.GetString(reader.GetAssemblyReference((AssemblyReferenceHandle)t.ResolutionScope).Name)
+                        : "";
```

Comment:
>"" [](start = 26, length = 2)

It feels like either we should be able to handle other kinds of ResolutionScopes, or shouldn't put this into the set of general helpers. Consider placing this method next to the tests that are using it.